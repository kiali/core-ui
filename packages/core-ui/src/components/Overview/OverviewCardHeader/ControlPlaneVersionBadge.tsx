import { Label } from '@patternfly/react-core';
import React from 'react';

export type ControlPlaneVersionBadgeProps = {
  version: string;
  isCanary?: boolean;
  style?: React.CSSProperties;
};

export const ControlPlaneVersionBadge = (props: ControlPlaneVersionBadgeProps) => {
  return (
    <Label style={props.style} color={props.isCanary ? 'blue' : 'orange'} isCompact>
      {props.version}
    </Label>
  );
};
