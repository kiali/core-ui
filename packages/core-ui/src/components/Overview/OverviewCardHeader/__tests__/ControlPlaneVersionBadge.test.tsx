import React from 'react';
import { render } from '@testing-library/react';

import { ControlPlaneVersionBadge } from '../ControlPlaneVersionBadge';

const mockControlPlaneVersionBadge = (version: string = '1.0', canary: boolean = true) => {
  return <ControlPlaneVersionBadge version={version} isCanary={canary} />;
};

describe('ControlPlaneVersionBadge', () => {
  test('Renders with canary', () => {
    const { asFragment, container, getByText } = render(mockControlPlaneVersionBadge('1.0', true));
    expect(asFragment()).toMatchSnapshot();
    expect(container.firstChild).toHaveClass('pf-m-blue');
    const versionElement = getByText('1.0');
    expect(versionElement).toBeVisible();
    expect(versionElement).toHaveClass('pf-c-label__content');
  });

  test('Renders without canary', () => {
    const { asFragment, container, getByText } = render(mockControlPlaneVersionBadge('1.0', false));
    expect(asFragment()).toMatchSnapshot();
    expect(container.firstChild).toHaveClass('pf-m-orange');
    const versionElement = getByText('1.0');
    expect(versionElement).toBeVisible();
    expect(versionElement).toHaveClass('pf-c-label__content');
  });
});
