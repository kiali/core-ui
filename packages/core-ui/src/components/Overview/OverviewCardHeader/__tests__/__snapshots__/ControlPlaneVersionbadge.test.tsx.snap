// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the snapshot 1`] = `
<DocumentFragment>
  <span
    class="pf-c-label pf-m-blue pf-m-compact"
  >
    <span
      class="pf-c-label__content"
    >
      1.0
    </span>
  </span>
</DocumentFragment>
`;
